### ISSUE TITLE
Unsuccessfully starting a stream confuses OnDeactivate()

### ISSUE THREAD

**Issue by WizardCM** [obsproject/obs-studio#3018]
A working connection goes `STARTING` -> `STARTED` -> bla bla bla -> `STOPPING` -> `STOPPED`.
However, a failed streaming attempt goes `STARTING` ->  `STOPPED`.
The problem is that when `stop` is called, the UI assumes `start` was called.

## Platform
Operating system and version: Windows 10 2004 (19041)
OBS Studio version: https://github.com/obsproject/obs-studio/commit/71506a8f722dabf3f71c0c61576ffdeb1d5b29ba (latest master at time of writing)

## Expected Behavior
`OnDeactivate` should clean up anything that `OnActivate` changed.

## Current Behavior
Instead, `OnDeactivate` can run without `OnActivate` being triggered, while incorrectly assuming `OnActivate` was called

OnActivate does this:
https://github.com/obsproject/obs-studio/blob/71506a8f722dabf3f71c0c61576ffdeb1d5b29ba/UI/window-basic-main.cpp#L5294-L5298

OnDeactivate does this:
https://github.com/obsproject/obs-studio/blob/71506a8f722dabf3f71c0c61576ffdeb1d5b29ba/UI/window-basic-main.cpp#L5319

**TL;DR:** It's possible for `outputHandler->Active()` to be false while `ui->profileMenu->isEnabled()` is true.

## Steps to Reproduce
1. Configure a YouTube stream with an invalid stream key
2. Attempt to go live
3. Notice `OnDeactivate` was called, and due to the incorrect state of `profileMenu`, it _enables_ certain features (recording icon in tray, etc) instead of attempting to disable them

## Additional information

While visually the only "bug" is seen in the tray icon and the taskbar, this also means that the sleep inhibition and process priority aren't cleared correctly.

The simplest fix I considered is removing the `profileMenu` check, though I don't know what side effects it may have. Refactoring how/when `OnActivate` is called may be a better solution.

**Comment by WizardCM** [comment]
As the side effects caused by this bug are mostly fixed in https://github.com/obsproject/obs-studio/commit/4869167f5c3161a3930d5f10b1778d1b53f97d0c, I'm closing this issue for now.
